sway-lsp/src/core/session.rs

```diff
@@ -330,6 +330,13 @@ impl Session {
         None
     }
 
+    /// Returns the [Namespace] from the compiled program if it exists.
+    pub fn namespace(&self) -> Option<Namespace> {
```

Comment:
Is it possible to return a reference to `Namespace` instead? Cloning the namespace is pretty heavy, would be nice to avoid if possible.

Reply:
Good catch!

Reply:
Would something like this work?

```rust
pub fn namespace<'a>(&'a self) -> Option<&'a Namespace> {
    let compiled_program = &*self.compiled_program.read();
    let program = compiled_program.typed.as_ref()?;
    Some(&program.root.namespace)
}
```

Reply:
With that I get this error: `cannot return value referencing temporary value`

I've tried a few different things but it seems like the function has to return the value for it to exist outside of the function.